opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -526,7 +526,7 @@ def _translate(self, record: logging.LogRecord) -> LogRecord:
             severity_text=record.levelname,
             severity_number=severity_number,
             body=body,
-            resource=self._logger.resource,
+            resource=getattr(self._logger, "resource", None),
```

Comment:
Would it be better to skip the `emit`-ting log record if the logger is no-op? https://github.com/open-telemetry/opentelemetry-python/blob/9ced357e9ffe32eabe03142848d0c206d026d441/opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py#L533-L539

Reply:
That could work but it would still technically break if user implements a logging sdk themselves. The current implementation of `_translate` requires the implemented handler to have a resource as an attribute of the logger, which it may not have (NoOp is one instance of that). I don't have too strong of an opinion on this though, just want to get this passing.

Reply:
I believe it's the implementer's responsibility to handle that. We have non-recording spans that get dropped. I was thinking along similar lines. Does it make sense to emit log records that don't have `Resource`?

Reply:
That's true. Will make this change to be in line with spans.